verification/builder.go

```diff
@@ -15,44 +15,67 @@ var (
 	certOidcIssuer           = "https://token.actions.githubusercontent.com"
 )
 
-var trustedReusableWorkflows = map[string]bool{
+var defaultTrustedReusableWorkflows = map[string]bool{
 	trustedBuilderRepository + "/.github/workflows/generator_generic_slsa3.yml": true,
 	trustedBuilderRepository + "/.github/workflows/builder_go_slsa3.yml":        true,
 }
 
 // VerifyWorkflowIdentity verifies the signing certificate information
-func VerifyWorkflowIdentity(id *WorkflowIdentity, source string) error {
+func VerifyWorkflowIdentity(id *WorkflowIdentity,
+	builderOpts *BuilderOpts, source string,
+) (string, error) {
 	// cert URI path is /org/repo/path/to/workflow@ref
 	workflowPath := strings.SplitN(id.JobWobWorkflowRef, "@", 2)
 	if len(workflowPath) < 2 {
-		return fmt.Errorf("%w: %s", errorMalformedWorkflowURI, id.JobWobWorkflowRef)
+		return "", fmt.Errorf("%w: workflow uri: %s", ErrorMalformedURI, id.JobWobWorkflowRef)
 	}
 
 	// Trusted workflow verification by name.
-	reusableWorkflowName := strings.Trim(workflowPath[0], "/")
-	if _, ok := trustedReusableWorkflows[reusableWorkflowName]; !ok {
-		return fmt.Errorf("%w: %s", ErrorUntrustedReusableWorkflow, reusableWorkflowName)
+	reusableWorkflowPath := strings.Trim(workflowPath[0], "/")
+	builderID, err := verifyTrustedBuilderID(reusableWorkflowPath, builderOpts.ExpectedID)
+	if err != nil {
+		return "", err
 	}
 
 	// Verify the ref.
 	if err := verifyTrustedBuilderRef(id, strings.Trim(workflowPath[1], "/")); err != nil {
-		return err
+		return "", err
 	}
 
 	// Issuer verification.
 	if !strings.EqualFold(id.Issuer, certOidcIssuer) {
```

Comment:
aside: Also curious about this. Is unicode case folding the way we should really be comparing issuers?

Reply:
great point. I'll update